addons/binding/org.openhab.binding.meterreader/src/main/java/org/openhab/binding/meterreader/internal/MeterReaderHandler.java

```diff
@@ -0,0 +1,292 @@
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.meterreader.MeterReaderBindingConstants;
+import org.openhab.binding.meterreader.MeterReaderConfiguration;
+import org.openhab.binding.meterreader.internal.helper.Baudrate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MeterReaderHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Matthias Steigenberger - Initial contribution
+ */
+public class MeterReaderHandler extends BaseThingHandler {
+
+    private static final int DEFAULT_REFRESH_PERIOD = 30;
+    private Logger logger = LoggerFactory.getLogger(MeterReaderHandler.class);
+    private MeterDevice<?> smlDevice;
+    private Cancelable valueReader;
+
+    public MeterReaderHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing SmlReader handler.");
+
+        MeterReaderConfiguration config = getConfigAs(MeterReaderConfiguration.class);
+        logger.debug("config port = {}", config.port);
+
+        boolean validConfig = true;
+        String errorMsg = null;
+
+        if (StringUtils.trimToNull(config.port) == null) {
+            errorMsg = "Parameter 'port' is mandatory and must be configured";
+            validConfig = false;
+        }
+
+        if (validConfig) {
+            try {
+                int baudrate = config.baudrate == null ? Baudrate.AUTO.getBaudrate()
+                        : Baudrate.fromString(config.baudrate).getBaudrate();
+                byte[] pullSequence = config.initMessage == null ? null
+                        : Hex.decodeHex(StringUtils.deleteWhitespace(config.initMessage).toCharArray());
+                this.smlDevice = MeterDeviceFactory.getDevice(config.mode, this.thing.getUID().getAsString(),
+                        config.port, pullSequence, baudrate, config.baudrateChangeDelay);
+            } catch (DecoderException e) {
+                logger.error("Failed to decode init message", e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Parameter 'initMessage' can not be decoded: " + e.getLocalizedMessage());
+            }
+            updateOBISValue();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+        }
+        updateStatus(ThingStatus.ONLINE);
```

Comment:
This thing goes always online even after ThingStatus.OFFLINE in line 84+89?

Reply:
Yeah, that was a bad mistake...
Thanks for finding it!